### ISSUE TITLE
Breaking lines before method calls does not work

### ISSUE THREAD

**Issue by huxoll** [tario/evalhook#6]
While I was working around #5 , I used a chained invocation to get one logical line.  The workaround was broken by the fix for #5.

The following generates a syntax error:

``` Ruby
require "shikashi"

class X
  def foo
    print "X#foo\n"
    return self
  end

  def bar
    print "X#bar\n"
    return self
  end
end

include Shikashi

to_run = <<-END
# Following 3 lines should do the same thing twice
X.new.foo().bar()
X.new.foo()
.bar()
END

s = Sandbox.new
priv = Privileges.new
priv.allow_method :print
priv.allow_const_read("X") # allow the access of X constant
priv.object(X).allow(:new)
priv.instances_of(X).allow_all

puts "Running block: #{to_run}"
s.run(priv, to_run, :no_base_namespace => true)
```

This is a minor bug for me, but chaining calls is useful for a DSL.

**Comment by tario** [comment]
This is not a minor bug, this basic code structure should work (unrelated note: the option no_base_namespace will be deprecated soon)

**Comment by tario** [comment]
I just tested this, breaking lines with the dots is NOT supported by the parser. I will edit the title of this issue

For example, this works for me:

``` Ruby
require "shikashi"

class X
  def foo
    print "X#foo\n"
    return self
  end

  def bar
    print "X#bar\n"
    return self
  end
end

include Shikashi

# avoid breaking lines before DOT!!
to_run = <<-END
# Following 3 lines should do the same thing twice
X.new.foo().bar()
X.new.foo().bar()
END

s = Sandbox.new
priv = Privileges.new
priv.allow_method :print
priv.allow_const_read("X") # allow the access of X constant
priv.object(X).allow(:new)
priv.instances_of(X).allow_all

puts "Running block: #{to_run}"
s.run(priv, to_run, :no_base_namespace => true)
```

**Comment by tario** [comment]
NOTES to fix this: update processors to use last RubyParser versions. RubyParser version 2.X.X does not support breaking lines on DOTs, buy latest versions of RubyParsers DO support, so, this will fixed when dependencies will be updated to latests versions of ruby_parser (code of tree processors should be updated)

The following code:

``` ruby
require "ruby_parser"

to_run = <<-END
x
.foo()
END

RubyParser.new.parse(to_run)
```

Fails with ruby_parser version 2.3.1, but works fine with ruby_parser 3.2.2

**Comment by tario** [comment]
Fixed on version 0.5.6. Use  _gem update evalhook_
